### PR TITLE
Added CentOS install script

### PR DESCRIPTION
**Pull request by MAHDTech**
Added a shell script that will take a minimal CentOS system and install IntelMQ and IntelMQ Manager

### CODE REVIEW

**Comment by aaronkaplan** [comment]
hi, I looked at it. 
While it looks like a neat script and while I really want to thank @MAHDTech for the contribution, I am wondering about one thing:
  * how can we have a unified installation mechanism ?
  * On Debian /Ubuntu we plan to provide .deb packages with proper release cycles
  * On CentOS it would be good to have .rpms which do exactly what @MAHDTech's script basically does.

Can we find a common, unified way?

**Comment by MAHDTech** [comment]
After doing some digging (never built rpm/deb before) there is a package "fpm" which is able to spit out both .deb.and .rpm from a single configuration.

I will look into this further, as perhaps it can then be some Jenkins? that auto-builds the latest, or some such.

Thoughts?

**Comment by aaronkaplan** [comment]
On Wed, May 11, 2016 at 09:19:10PM -0700, MAHDTech wrote:
> After doing some digging (never built rpm/deb before) there is a package "fpm" which is able to spit out both .deb.and .rpm from a single configuration.
> 
> I will look into this further, as perhaps it can then be some Jenkins? that auto-builds the latest, or some such.
> 
> Thoughts?

sounds interesting. Did you look at
https://github.com/certtools/intelmq/pull/507 ?

**Comment by sebix** [comment]
And there's the [OpenSUSE Build Service](https://build.opensuse.org/) which allows to build (and test) debs for Ubuntu and Debian (and rpms for many others) from specfiles. I successfully build some rpms with it, but haven't done any debs yet.

**Comment by SYNchroACK** [comment]
@sebix IMHO we can merge this one and continue the discussion about the packaging in a issue. Lets merge it ;)

**Comment by sebix** [comment]
> @sebix IMHO we can merge this one and continue the discussion about the packaging in a issue. Lets merge it ;)

Have you done a review and tested it?

File: docs/install-scripts/Install-IntelMQ-CentOS.sh [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    # tail -f /var/log/secure
+    # Since SELinux is enabled we need to allow httpd to write to certain directories
+    chcon -R -t httpd_sys_content_rw_t /opt/intelmq/var 2>&1 >> $LOG
+    chcon -R -t httpd_sys_content_rw_t /var/www/intelmq-manager/php 2>&1 >> $LOG
+
+    # Allow httpd mod auth
+    setsebool -P allow_httpd_mod_auth_pam 1 2>&1 >> $LOG
+
+    # Allow httpd to connect to the redis server over tcp/ip
+    setsebool -P httpd_can_network_connect on 2>&1 >> $LOG
```
Comment (sebix) [inline docs/install-scripts/Install-IntelMQ-CentOS.sh]: We does httpd need to connect to redis?

File: docs/install-scripts/Install-IntelMQ-CentOS.sh [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    # Set the SELinux exceptions to allow Apache to
+    # work properly with IntelMQ Manager
+
+    # Install the Audit2Allow and Audit2Why so we can get IntelMQ working with SELinux
+    # these tools are very helpful with SELinux
+    yum -y install policycoreutils-python 2>&1 >> $LOG
+
+    # tail -f /var/log/secure
+    # Since SELinux is enabled we need to allow httpd to write to certain directories
+    chcon -R -t httpd_sys_content_rw_t /opt/intelmq/var 2>&1 >> $LOG
```
Comment (sebix) [inline docs/install-scripts/Install-IntelMQ-CentOS.sh]: Also needed: `/opt/intelmq/etc`

File: docs/install-scripts/Install-IntelMQ-CentOS.sh [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  chmod 0440 /etc/sudoers.d/intelmq 2>&1 >> $LOG
+
+  if [[ $DISABLE_SELINUX == "Y" ]]
+  then
+
+    echo $RED"Taking the easy way out hey? I'm sure there's a pill for that"$NC
+    echo -e "\n"
+
+    # Set SELINUX to disabled
+    sed -i 's|SELINUX=enforcing|SELINUX=disabled|g' /etc/selinux/config
```
Comment (sebix) [inline docs/install-scripts/Install-IntelMQ-CentOS.sh]: Do we need to restart/reload selinux here?

**Comment by sebix** [comment]
For selinux disabled I can confirm it works. In enabled state, I couldn't yet figure out why the manager can't receive data from ctl.

File: docs/install-scripts/Install-IntelMQ-CentOS.sh [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+  echo -e "\n" | tee -a $LOG
+  echo -e $RED"Installing IntelMQ Manager"$NC | tee -a $LOG
+  echo -e "\n" | tee -a $LOG
+
+  # Install the requirements for the web interface
+  yum -y install httpd mod_ssl php 2>&1 >> $LOG
+
+  # Allow the ports in the firewall and reload the configuration to apply
+  sudo firewall-cmd --permanent --add-port=$PORT/tcp
```
Comment (sebix) [inline docs/install-scripts/Install-IntelMQ-CentOS.sh]: This command is not available on a fresh installation of centos 7.

**Comment by SYNchroACK** [comment]
Found a repository with related code:
https://github.com/mariuszlitwin/intelmq_bootstrap/blob/master/bootstrap.sh

**Comment by wagner-certat** [comment]
@SYNchroACK this is for debian/ubuntu

**Comment by wagner-certat** [comment]
Closing this now as we have both packages and install instruction in the docs and the selinux part of this script is not working anyway.

**Pull request closed without merging**